tempto-core/src/test/groovy/io/trino/tempto/internal/initialization/TestInitializationListenerTest.groovy

```diff
@@ -189,6 +196,29 @@ class TestInitializationListenerTest
         return TestClass.getMethod('testMethodFailed')
     }
 
+    static class TestClassNoOverrideAnnotatedMethods
+            extends TestClass
+    {
+    }
+
+    static class TestClassOverrideAnnotatedMethods
+            extends TestClass
+    {
+        @Override
+        @BeforeTestWithContext
```

Comment:
Do we need to annotate it once again? Why not to take annotation from `TestClass`? Does it change the order in which methods are invoked? Consider a case:

```
class B { @BeforeTestWithContext void b() {} }
class A extends B { @BeforeTestWithContext void a() {} }
```

vs 

```
class B { @BeforeTestWithContext void b() {} }
class A extends B { @BeforeTestWithContext void a() {} void b() {} }
```

Will `b()` be called always as first?

Reply:
Yes b() will be always called first, as it will be always returned first from getMethods() call.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Class.html#getMethods()

Reply:
Should we have test for that?

Reply:
Just added: TestClassOverrideAdditionalAnnotatedMethod :)